common-pg/src/main/scala/com/socrata/pg/analyzer2/SoQLFunctionSqlizerRedshift.scala

```diff
@@ -0,0 +1,729 @@
+package com.socrata.pg.analyzer2
+
+import com.socrata.prettyprint.prelude._
+import com.socrata.soql.analyzer2._
+import com.socrata.soql.collection.NonEmptySeq
+import com.socrata.soql.types._
+import com.socrata.soql.functions.SoQLFunctions._
+import com.socrata.soql.functions.{Function, MonomorphicFunction, SoQLTypeInfo}
+import SoQLFunctionSqlizerRedshift._
+
+class SoQLFunctionSqlizerRedshift[MT <: MetaTypes with ({ type ColumnType = SoQLType; type ColumnValue = SoQLValue })] extends FuncallSqlizer[MT] {
+  import SoQLTypeInfo.hasType
+
+  implicit class ExprInterpolator(sc: StringContext) {
+    def expr(args: Any*): OrdinaryFunctionSqlizer =
+      ofs { (f, runtimeArgs, ctx) =>
+        assert(args.length >= f.function.minArity)
+        assert(f.function.allParameters.startsWith(runtimeArgs.map(_.typ)))
+
+        val expandedRuntimeArgs = args.map(_.asInstanceOf[Int]).map(n => runtimeArgs.toList(n))
+
+      val sql = Doc(sc.s(expandedRuntimeArgs.map(_.compressed.sql): _*))
+      ExprSql(sql, f)
+    }
+  }
+
+
+  def wrap(e: Expr, exprSql: ExprSql, wrapper: String, additionalWrapperArgs: Doc*) =
+    ExprSql((exprSql.compressed.sql +: additionalWrapperArgs).funcall(Doc(wrapper)), e)
+
+  def comment(sqlizer: OrdinaryFunctionSqlizer, comment: String) = ofs { (f, args, ctx) =>
+    val e = sqlizer(f, args, ctx)
+    ExprSql((d"/*" +#+ Doc(comment) +#+ Doc("*/") +#+ e.compressed.sql).parenthesized, f)
+  }
```

Comment:
Use an annotation for this (add a new subclass to SqlizeAnnotation, and add it to a Doc with `.ann(YourNewAnnotation(whatever))` and then teach the pretty sql renderer over in ProcessQuery about it.

Reply:
An Annotation would mean this *does not* show up in the generated SQL, but instead the annotated generated SQL when including debug info, right?

Reply:
Right

Reply:
The idea was to keep the generated-SQL greppable and so to have the commentshow up in the generated SQL.
Happy to remove it if you think it's messy.